### DISCUSSION TITLE
How can I set width to be "container" in config?

### DISCUSSION THREAD

**Discussion by ghost** [vega/altair#3352] [Q&A] [answered]
It seems to me I can only set width as "container" in the toplevel, but not in `config.view`

**Comment by ghost** [comment]
Another related question is, in the doc: https://altair-viz.github.io/user_guide/customization.html#defining-a-custom-theme
<img width="275" alt="image" src="https://github.com/altair-viz/altair/assets/107285219/9aa2c500-ca23-4056-91bd-329bf001bcb4">
We can set "width" in `config.view`.

However, in vega-lite documentation: https://vega.github.io/vega-lite/docs/spec.html#config.  There is no "width" attribute, the only thing is "continuousWidth" and "discreteWidth".

**Comment by joelostblom** [comment, marked as answer]
You can set `'container'` at the top-level rather than inside the `config.view` key as per https://vega.github.io/vega-lite/docs/size.html#specifying-responsive-width-and-height. 

You're right that the example in the altair docs is using an unsupported spec (although it still works), it should look like this (feel free to submit a PR to the docs): 

```py
def black_marks():
    return {
        'height': 200,
        'width': 200,
        'config': {
            'mark': {
                'color': 'black',
                'fill': 'black'
            }
        }
    }
```
